MetadataExtractor/Formats/Xmp/XmpDirectory.cs

```diff
@@ -76,17 +78,43 @@ protected override bool TryGetTagName(int tagType, out string tagName)
         [NotNull]
         public IDictionary<string, string> GetXmpProperties()
         {
-            return XmpMeta?.Properties
-                       .Where(p => p.Path != null)
-                       .ToDictionary(p => p.Path, p => p.Value)
-                   ?? new Dictionary<string, string>();
+            var propertyValueByPath = new Dictionary<string, string>();
+            if (XmpMeta != null)
+            {
+                try
+                {
+                    XmpIterator i = new XmpIterator((XmpMeta)XmpMeta, null, null, new IteratorOptions() { IsJustLeafNodes = true });
```

Comment:
(Wow, `XmpIterator` has an ugly API!)

You could cache an instance of `IteratorOptions` somewhere static if you wanted, to reduce allocations.

Reply:
Also done